AlphaWallet/Tokens/Coordinators/ClaimOrderCoordinator.swift

```diff
@@ -22,23 +22,52 @@ class ClaimOrderCoordinator {
         self.web3 = web3
     }
 
-    func claimOrder(indices: [UInt16],
+    func claimOrder(signedOrder: SignedOrder,
                     expiry: BigUInt,
                     v: UInt8,
                     r: String,
                     s: String,
                     completion: @escaping (Result<String, AnyError>) -> Void
         ) {
-        let request = ClaimERC875Order(expiry: expiry, indices: indices, v: v, r: r, s: s)
+
+        if let tokenIds = signedOrder.order.tokenIds {
```

Comment:
I think the conditions here has to be tweaked a bit because `signedOrder.order.tokenIds` might be `nil`. I suppose when it is `nil`, we want it to execute the `else` too? If that's what we want, then you can change this:

    if let tokenIds = signedOrder.order.tokenIds {
        if !tokenIds.isEmpty {

to:

    if let tokenIds = signedOrder.order.tokenIds, if !tokenIds.isEmpty {
        //the if's body as it was for spawable
    } else {
        //the else's body as it was for non-spawable
    }

Also, maybe better to extract another method, say, `claimNonSpawnableOrder()`? So it's like this:

    func claimOrder(...) {
        if let tokenIds = signedOrder.order.tokenIds, if !tokenIds.isEmpty {
            claimSpawnableOrder()
        } else {
            claimNonSpawnableOrder()
        }
    }

    func claimNonSpawnableOrder(...) {
        //What was in else's body
    }

Reply:
Maybe as a future refactoring, we can also create something like:

enum OrderType {
    case spawnable
    case nonSpawnable
}

and stuff some variables or even logic inside. The usual strong-typing stuff.